scripts/bert/staticbert/static_finetune_squad.py

```diff
@@ -0,0 +1,533 @@
+            new_lr = lr * step_num / num_warmup_steps
+        else:
+            offset = (step_num - num_warmup_steps) * lr / \
+                     (num_train_steps - num_warmup_steps)
+            new_lr = lr - offset
+        trainer.set_learning_rate(new_lr)
+        return step_num
+
+    # Do not apply weight decay on LayerNorm and bias terms
+    for _, v in net.collect_params('.*beta|.*gamma|.*bias').items():
+        v.wd_mult = 0.0
+    # Collect differentiable parameters
+    params = [p for p in net.collect_params().values()
+              if p.grad_req != 'null']
+    # Set grad_req if gradient accumulation is required
+    if accumulate:
+        for p in params:
+            p.grad_req = 'add'
+
+    epoch_tic = time.time()
+    total_num = 0
+    log_num = 0
+    for epoch_id in range(epochs):
+        step_loss = 0.0
+        tic = time.time()
+        for batch_id, data in enumerate(train_dataloader):
+            # set new lr
+            step_num = set_new_lr(step_num, batch_id)
+            # forward and backward
+            with mx.autograd.record():
+                _, inputs, token_types, valid_length, start_label, end_label = data
+
+                log_num += len(inputs)
+                total_num += len(inputs)
+
+                out = net(inputs.astype('float32').as_in_context(ctx),
+                          token_types.astype('float32').as_in_context(ctx),
+                          valid_length.astype('float32').as_in_context(ctx))
+
+                ls = loss_function(out, [
+                    start_label.astype('float32').as_in_context(ctx),
+                    end_label.astype('float32').as_in_context(ctx)]).mean()
+
+                if accumulate:
+                    ls = ls / accumulate
+            ls.backward()
+            # update
+            if not accumulate or (batch_id + 1) % accumulate == 0:
+                trainer.allreduce_grads()
+                nlp.utils.clip_grad_global_norm(params, 1)
+                trainer.update(1)
+
+            step_loss += ls.asscalar()
+
+            if (batch_id + 1) % log_interval == 0:
+                toc = time.time()
+                log.info(
+                    'Epoch: {}, Batch: {}/{}, Loss={:.4f}, lr={:.7f} Time cost={:.1f} Thoughput={:.2f} samples/s'  # pylint: disable=line-too-long
+                        .format(epoch_id, batch_id, len(train_dataloader),
+                                step_loss / log_interval,
+                                trainer.learning_rate, toc - tic, log_num / (toc - tic)))
+                tic = time.time()
+                step_loss = 0.0
+                log_num = 0
+        epoch_toc = time.time()
+        log.info('Epoch: {}, Time cost={:.2f} s, Thoughput={:.2f} samples/s'
+                 .format(epoch_id, epoch_toc - epoch_tic,
+                         len(train_dataloader) / (epoch_toc - epoch_tic)))
+
+    net.save_parameters(os.path.join(output_dir, 'net.params'))
+
+
+def evaluate():
```

Comment:
Evaluating the HybridBlock version of the BERT model does not add much value in educating users, especially given that this is already avaiable in the other static_bert_qa script. If you think showing the evaluation is useful, consider making the evaluation optional, and use symbol block with the exported symbol and parameters instead.